src/AppBundle/Entity/Report/Report.php

```diff
@@ -878,20 +890,19 @@ public function setAgreedBehalfDeputyExplanation($agreedBehalfDeputyExplanation)
         return $this;
     }
 
-    public function isDue()
+    /**
+     * @param \DateTime|null $endDate
+     * @return bool
+     */
+    public static function isDue(\DateTime $endDate = null)
```

Comment:
why static? A report must have an end date from when its created. The only thing that end date is compared against is the current date which we will always know.

Reply:
I need to reuse the same "midnight" logic from the report after the array hydration. It just depends on the end date and related to report, needed in services not connected, so I placed it there. better solution in mind ?